syscalls_darwin.go

```diff
@@ -151,6 +151,11 @@ func (t *tunReadCloser) Read(to []byte) (int, error) {
 }
 
 func (t *tunReadCloser) Write(from []byte) (int, error) {
+
+	if from == nil || len(from) == 0 {
```

Comment:
You don't need to check `from == nil`. `len(from) == 0` covers the `nil` case too :)